tracer/src/Datadog.Trace/Iast/InsecureAuthAnalyzer.cs

```diff
@@ -0,0 +1,154 @@
+// <copyright file="InsecureAuthAnalyzer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Linq;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+#if !NETFRAMEWORK
+using Microsoft.AspNetCore.Http;
+using Microsoft.Net.Http.Headers;
+#endif
+
+namespace Datadog.Trace.Iast;
+
+#nullable enable
+
+internal static class InsecureAuthAnalyzer
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(InsecureAuthAnalyzer));
+
+#if !NETFRAMEWORK
+    public static void AnalyzeInsecureAuth(IHeaderDictionary headers, IntegrationId integrationId, int statusCode)
+    {
+        try
+        {
+            if (IgnoredStatusCode(statusCode))
+            {
+                return;
+            }
+
+            // Check for Authorization header
+            const string headerName = HeaderNames.Authorization;
+            if (!headers.TryGetValue(headerName, out var authHeader) || string.IsNullOrEmpty(authHeader))
```

Comment:
Calling `string.IsNullOrEmpty` implicitly converts the `StringValues` to `string`, which will result in allocation if there are multiple values. Instead, use the `StringValues.IsNullOrEmpty()` method 🙂 

```suggestion
            if (!headers.TryGetValue(headerName, out var authHeader) || StringValues.IsNullOrEmpty(authHeader))
```

Reply:
fixed in 3507fbd